src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -971,50 +971,58 @@ LRESULT CALLBACK FancyZones::s_WndProc(HWND window, UINT message, WPARAM wparam,
 
 void FancyZones::UpdateZoneWindows() noexcept
 {
+    // Mapping between display device name and device index (operating system identifies each display device in the current session with an index value).
+    std::unordered_map<std::wstring, DWORD> displayDeviceIdx;
+    struct capture
+    {
+        FancyZones* fancyZones;
+        std::unordered_map<std::wstring, DWORD>* displayDeviceIdx;
+    };
+
     auto callback = [](HMONITOR monitor, HDC, RECT*, LPARAM data) -> BOOL {
+        capture* params = reinterpret_cast<capture*>(data);
         MONITORINFOEX mi;

```

Comment:
let's initialize `mi` the same way as we do `displayDevice`

Reply:
done